Index: sagemaker-code-editor/vscode/src/vs/workbench/services/userDataProfile/browser/globalStateResource.ts
===================================================================
--- sagemaker-code-editor.orig/vscode/src/vs/workbench/services/userDataProfile/browser/globalStateResource.ts
+++ sagemaker-code-editor/vscode/src/vs/workbench/services/userDataProfile/browser/globalStateResource.ts
@@ -20,6 +20,21 @@ interface IGlobalState {
 	storage: IStringDictionary<string>;
 }
 
+interface IViewlet {
+	id: string;
+	pinned: boolean;
+	visible: boolean;
+	order: number;
+}
+
+const VIEWLET_AMAZONQ = 'workbench.view.extension.amazonq';
+const VIEWLET_AWSTOOLKIT = 'workbench.view.extension.aws-explorer';
+
+const AI_LEAGUE_VIEWLETS = [
+	VIEWLET_AMAZONQ,
+	VIEWLET_AWSTOOLKIT,
+];
+
 export class GlobalStateResourceInitializer implements IProfileResourceInitializer {
 
 	constructor(@IStorageService private readonly storageService: IStorageService) {
@@ -27,7 +42,20 @@ export class GlobalStateResourceInitiali
 
 	async initialize(content: string): Promise<void> {
 		const globalState: IGlobalState = JSON.parse(content);
+		const pinnedViewlets = JSON.parse(globalState.storage['workbench.activity.pinnedViewlets2']);
+
+		const newPinnedViewlets = pinnedViewlets.map((viewlet: IViewlet) => (
+			{
+				...viewlet,
+				pinned: AI_LEAGUE_VIEWLETS.includes(viewlet.id),
+				visible: viewlet.id === VIEWLET_AWSTOOLKIT,
+			}
+		));
+
+		globalState.storage['workbench.activity.pinnedViewlets2'] = JSON.stringify(newPinnedViewlets);
+
 		const storageKeys = Object.keys(globalState.storage);
+
 		if (storageKeys.length) {
 			const storageEntries: Array<IStorageEntry> = [];
 			for (const key of storageKeys) {
@@ -53,8 +81,6 @@ export class GlobalStateResource impleme
 	}
 
 	async apply(content: string, profile: IUserDataProfile): Promise<void> {
-		console.log('globalState profile:', profile);
-		console.log('globalState content:', content);
 		const globalState: IGlobalState = JSON.parse(content);
 		await this.writeGlobalState(globalState, profile);
 	}
@@ -72,7 +98,6 @@ export class GlobalStateResource impleme
 
 	private async writeGlobalState(globalState: IGlobalState, profile: IUserDataProfile): Promise<void> {
 		const storageKeys = Object.keys(globalState.storage);
-		console.log('global state:', profile);
 		if (storageKeys.length) {
 			const updatedStorage = new Map<string, string | undefined>();
 			const nonProfileKeys = [
Index: sagemaker-code-editor/vscode/src/vs/workbench/browser/parts/paneCompositeBar.ts
===================================================================
--- sagemaker-code-editor.orig/vscode/src/vs/workbench/browser/parts/paneCompositeBar.ts
+++ sagemaker-code-editor/vscode/src/vs/workbench/browser/parts/paneCompositeBar.ts
@@ -83,6 +83,11 @@ export interface IPaneCompositeBarOption
 	readonly colors: (theme: IColorTheme) => ICompositeBarColors;
 }
 
+const AI_LEAGUE_VIEWLETS = [
+	'workbench.view.extension.aws-explorer',
+	'workbench.view.extension.amazonq',
+];
+
 export class PaneCompositeBar extends Disposable {
 
 	private readonly viewContainerDisposables = this._register(new DisposableMap<string, IDisposable>());
@@ -578,12 +583,28 @@ export class PaneCompositeBar extends Di
 
 	private storeCachedViewContainersState(cachedViewContainers: ICachedViewContainer[]): void {
 		const pinnedViewContainers = this.getPinnedViewContainers();
-		this.setPinnedViewContainers(cachedViewContainers.map(({ id, pinned, order }) => ({
+		const views = cachedViewContainers.map(({ id, pinned, order }) => ({
 			id,
 			pinned,
 			visible: Boolean(pinnedViewContainers.find(({ id: pinnedId }) => pinnedId === id)?.visible),
 			order
-		} satisfies IPinnedViewContainer)));
+		} satisfies IPinnedViewContainer));
+
+		const hideViews = views.map((view) => {
+			const shouldHide = view.id.startsWith('workbench.view') && !AI_LEAGUE_VIEWLETS.includes(view.id);
+
+			return {
+				...view,
+				pinned: !shouldHide,
+				visible: !shouldHide,
+			};
+		});
+
+		if (this.location === ViewContainerLocation.Sidebar) {
+			this.setPinnedViewContainers(hideViews);
+		} else {
+			this.setPinnedViewContainers(views);
+		}
 
 		this.setPlaceholderViewContainers(cachedViewContainers.map(({ id, icon, name, views, isBuiltin }) => ({
 			id,
