Index: sagemaker-code-editor/vscode/src/vs/code/browser/workbench/workbench.ts
===================================================================
--- sagemaker-code-editor.orig/vscode/src/vs/code/browser/workbench/workbench.ts
+++ sagemaker-code-editor/vscode/src/vs/code/browser/workbench/workbench.ts
@@ -478,6 +478,10 @@ class WorkspaceProvider implements IWork
 
 	private createTargetUrl(workspace: IWorkspace, options?: { reuse?: boolean; payload?: object }): string | undefined {
 
+		// Preserve origin query parameter if it exists
+		const currentUrl = new URL(document.location.href);
+		const originParam = currentUrl.searchParams.get('origin');
+
 		// Empty
 		let targetHref: string | undefined = undefined;
 		if (!workspace) {
@@ -496,6 +500,11 @@ class WorkspaceProvider implements IWork
 			targetHref = `${document.location.origin}${document.location.pathname}?${WorkspaceProvider.QUERY_PARAM_WORKSPACE}=${queryParamWorkspace}`;
 		}
 
+		// Append origin parameter if it exists
+		if (originParam && targetHref) {
+			targetHref += `&origin=${encodeURIComponent(originParam)}`;
+		}
+
 		// Append payload if any
 		if (options?.payload) {
 			targetHref += `&${WorkspaceProvider.QUERY_PARAM_PAYLOAD}=${encodeURIComponent(JSON.stringify(options.payload))}`;
